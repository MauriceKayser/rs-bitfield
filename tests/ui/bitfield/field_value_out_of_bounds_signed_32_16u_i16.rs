//! Assertion failure: Field should be signed.

extern crate alloc;

#[bitfield::bitfield(32)]
struct BitField(#[field(size = 16)] Field);

#[derive(Clone, Copy, Debug, bitfield::Field)]
#[repr(i16)]
enum Field {
    F2 = -300
}

fn main() {}