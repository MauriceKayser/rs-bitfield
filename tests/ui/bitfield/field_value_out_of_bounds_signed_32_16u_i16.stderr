error[E0271]: type mismatch resolving `<Field as TryFrom<u16>>::Error == u16`
 --> tests/ui/bitfield/field_value_out_of_bounds_signed_32_16u_i16.rs:6:37
  |
6 | struct BitField(#[field(size = 16)] Field);
  |                                     ^^^^^ expected `u16`, found `Infallible`

error[E0277]: the trait bound `Field: TryFrom<u16>` is not satisfied
 --> tests/ui/bitfield/field_value_out_of_bounds_signed_32_16u_i16.rs:6:32
  |
6 | struct BitField(#[field(size = 16)] Field);
  |                                ^^^^^-----
  |                                |    |
  |                                |    required by a bound introduced by this call
  |                                the trait `From<u16>` is not implemented for `Field`
  |
  = help: the trait `TryFrom<u16>` is not implemented for `Field`
          but trait `TryFrom<i16>` is implemented for it
  = help: for that trait implementation, expected `i16`, found `u16`
  = note: required for `u16` to implement `Into<Field>`
  = note: required for `Field` to implement `TryFrom<u16>`

error[E0277]: the trait bound `Field: From<u16>` is not satisfied
 --> tests/ui/bitfield/field_value_out_of_bounds_signed_32_16u_i16.rs:6:37
  |
6 | struct BitField(#[field(size = 16)] Field);
  |                                     ^^^^^ the trait `From<u16>` is not implemented for `Field`
  |
  = help: the trait `TryFrom<u16>` is not implemented for `Field`
          but trait `TryFrom<i16>` is implemented for it
  = help: for that trait implementation, expected `i16`, found `u16`
  = note: required for `u16` to implement `Into<Field>`
  = note: required for `Field` to implement `TryFrom<u16>`
